### PR TITLE
add helper method to add ephemeral messages; also switch 'delete mess…

### PR DESCRIPTION
**Pull request by florianorben**
…age' to this new method

Allows to add ephemeral messages to post store without storing them in the backend.
```javascript
PostStore.addEphemeralPost(post, [notify], [autoDelete]);
```

### CODE REVIEW

**Comment by ninja-** [comment]
what's the usecase?

**Comment by florianorben** [comment]
automated messaged that do not need to be stored in the backend, like "message deleted"

**Comment by ninja-** [comment]
but wasn't it supposed to stay as <message deleted> forever?

**Comment by florianorben** [comment]
No, only temporarily, for a short while after the message has been deleted

File: web/react/dispatcher/event_helpers.jsx [hunk 1]
```diff
@@ -148,3 +148,8 @@ export function emitClearSuggestions(suggestionId) {
         id: suggestionId
     });
 }
+
+// convenience method to dispatch an event in JS' next event cycle
+export function defer(fn) {
```
Comment (crspeller) [inline web/react/dispatcher/event_helpers.jsx]: This function would be better in utils.jsx. This file is for functions that create events for the dispacher. The "action creators" in this flux diagram: https://facebook.github.io/react/img/blog/flux-diagram.png  We are going to change the name to "action creators" soon so it makes more sense.

**Comment by jwilander** [comment]
+1 for the most part. My only real issue is that `AsyncClient.getChannels()` when switching channels could override the artificial count you create in `AsyncClient.getChannelAndAddUnreadMessages()`. If that's a non-issue then looks good

**Comment by florianorben** [comment]
@crspeller Makes sense, code updated!

@jwilander The way I see it (and what happend in manual tests) is that the count only would be overriden if you
- leave the channel, in that case doesnt matter
- you reload the page, then the message would be lost anyway :)
- or if a new post comes in from socketstore (which would then add to the unread count, so the unread label wouldn't be removed)

There could be some abstract(?) cases I havent thought about where the count would be overriden, but I haven't encountered any issues with the normal chatting/joining/leaving etc. testing/workflow :)

**Comment by jwilander** [comment]
Sounds good, +1

**Comment by crspeller** [comment]
+1

**Comment by crspeller** [comment]
@florianorben Needs rebase.

**Comment by florianorben** [comment]
@crspeller Rebased!

**Pull request merged**